utils/preferences/tests/dtf/mod.rs

```diff
@@ -0,0 +1,84 @@
+mod data_provider;
+mod options;
+mod preferences;
+
+use data_provider::get_defaults;
+use icu_datetime::options::length;
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{extensions_unicode_key, LanguageIdentifier, Locale};
+use icu_preferences::{preferences, Preferences};
+use options::{DayPeriod, LocaleMatcher};
+use preferences::{Calendar, NumberingSystem};
+
+preferences!(
+    DTFPreferencesBag,
+    DTFDefaultPreferencesBag,
+    DTFResolvedPreferencesBag,
+    {
+        hour_cycle => Option<HourCycle>, HourCycle, Some(extensions_unicode_key!("hc")),
+        calendar => Option<Calendar>, Calendar, Some(extensions_unicode_key!("ca")),
+        numbering_system => Option<NumberingSystem>, NumberingSystem, Some(extensions_unicode_key!("nu"))
+    },
+    DTFOptionsBag,
+    {
+        date_length => Option<length::Date>,
+        time_length => Option<length::Time>,
+        day_period => Option<DayPeriod>,
+        locale_matcher => Option<LocaleMatcher>,
+        time_zone => Option<bool>
+    }
+);
+
+pub struct DateTimeFormat {
+    prefs: DTFResolvedPreferencesBag,
+    options: Option<DTFOptionsBag>,
+}
+
+impl DateTimeFormat {
+    pub fn new(prefs: DTFPreferencesBag, options: Option<DTFOptionsBag>) -> Self {
```

Comment:
Nit: I think I prefer a non-optional `options` where you put `Default::default()` if you don't care. I don't want to argue about which is the common case (custom options or all defaults); I think we should treat them equally. Forcing custom options users to have `Some(...)` is more annoying, in my opinion, than `Default::default()` in the default-options case. I think it's more clear than `None` that you are setting things to the default.

Reply:
I'm slightly concerned about the size of Options::default() that you'd have to each time construct, but we can revisit it in 2.0.

Reply:
Thank you. Note that `Option<DateTimeFormatOptions>` is at least as big as `DateTimeFormatOptions`.